lib/macho/load_commands.rb

```diff
@@ -399,6 +339,18 @@ def initialize(view, cmd, cmdsize, name, timestamp, current_version, compatibili
       @current_version = current_version
       @compatibility_version = compatibility_version
     end
+
+    # @param endianness [Symbol, nil] the endianness of the serialized string,
+    #  or nil to use the view's endianness
+    # @return [String] the serialized fields of the load command
+    # @api private
+    def serialize(endianness = nil)
+      endianness ||= view.endianness
```

Comment:
*I'm commenting on this specific `serialize` method, but of course everything also applies to all the other `serialize` methods.*

I think we need to pass something that knows about the endianness and bit width of the Mach-O file this load command will be inserted into. To me, the easiest solution seems to be to just pass the `MachOFile` instance for querying this information (though we might want to abstract this away in the future). I also think we shouldn't even try to query the `view` for this information. It won't typically exist.